win32/win32-fetch.c

```diff
@@ -0,0 +1,39 @@
+// Fetch URL content to a specified handle using Wininet as no libcurl
+// dev libs available for Win32/64.
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+#include <wininet.h>
+#include <stdio.h>
+#include "win32-fetch.h"
+
+int win32_fetch(char *url, FILE *file, char *proxy, char *ua)
+{
+    HINTERNET hInt;
+    if (proxy != NULL)
+        hInt = InternetOpen(ua, INTERNET_OPEN_TYPE_PROXY, proxy, "<local>", 0);
+    else
+        hInt = InternetOpen(ua, INTERNET_OPEN_TYPE_DIRECT, NULL, NULL, 0);
+    if (NULL == hInt)
+        return 0x80000000 | (int)GetLastError();
+
+    HINTERNET hUrl = InternetOpenUrl(hInt, url, NULL, 0, INTERNET_FLAG_RELOAD, 0);
```

Comment:
This is a neat replacement for libcurl. But could you adapt it to compile with a C-89 compiler?

Reply:
May I ask why we need that level of compatibility?
I wish to enable more gcc checks and standard compliance once I'm done with Gtk+ 3 migration, but I was thinking about C11. So, now I am wondering why C-89 is better?

Reply:
Not sure why we would aim for C-89 either, but happy to move my variable declarations to the top if it helps.